src/client/Microsoft.Identity.Client/Http/HttpRetryCondition.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net;
+
+namespace Microsoft.Identity.Client.Http
+{
+    internal static class HttpRetryConditions
```

Comment:
Have you seen @Robbie-Microsoft 's similar abstraction - https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/4d3c84b3f5b633b269e46e03382fa8d834886ab0/lib/msal-node/src/retry/LinearRetryPolicy.ts#L7